pkg/login/social/generic_oauth.go

```diff
@@ -142,23 +142,21 @@ func (s *SocialGenericOAuth) UserInfo(ctx context.Context, client *http.Client,
 			}
 		}
 
-		if userInfo.Role == "" {
-			if !s.skipOrgRoleSync {
-				role, grafanaAdmin := s.extractRoleAndAdmin(data.rawJSON, []string{}, true)
-				if role != "" {
-					s.log.Debug("Setting user info role from extracted role")
-
-					userInfo.Role = role
-					if s.allowAssignGrafanaAdmin {
-						userInfo.IsGrafanaAdmin = &grafanaAdmin
-					}
-				}
+		if (userInfo.Role == "" || userInfo.Role == s.defaultRole()) && !s.skipOrgRoleSync {
+			role, grafanaAdmin, err := s.extractRoleAndAdmin(data.rawJSON, []string{}, true)
```

Comment:
This will fail if we fail to get the role from the token and `roleAttributeStrict`, right? I think we should allow for the role to be extracted from the API.